java.lang.NullPointerException: Cannot invoke "com.osgifx.console.supervisor.Supervisor.removeOSGiEventListener

```
14.11.2022 17:25:45.371 *ERROR* [Equinox Log Thread - Equinox Container: 67c0969b-c1f6-42e4-ad9a-e55b7b1069b3] LogService.org.eclipse.e4.ui.workbench Unable to create class 'com.osgifx.console.ui.events.handler.EventReceiveMenuContributionHandler' from bundle '103' (org.eclipse.e4.core.di.InjectionException: java.lang.NullPointerException: Cannot invoke "com.osgifx.console.supervisor.Supervisor.removeOSGiEventListener(com.osgifx.console.supervisor.EventListener)" because "this.supervisor" is null)
org.eclipse.e4.core.di.InjectionException: java.lang.NullPointerException: Cannot invoke "com.osgifx.console.supervisor.Supervisor.removeOSGiEventListener(com.osgifx.console.supervisor.EventListener)" because "this.supervisor" is null
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:68)
	at org.eclipse.e4.core.internal.di.InjectorImpl.processAnnotated(InjectorImpl.java:995)
	at org.eclipse.e4.core.internal.di.InjectorImpl.internalInject(InjectorImpl.java:140)
	at org.eclipse.e4.core.internal.di.InjectorImpl.internalMake(InjectorImpl.java:403)
	at org.eclipse.e4.core.internal.di.InjectorImpl.make(InjectorImpl.java:330)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.make(ContextInjectionFactory.java:202)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.createFromBundle(ReflectionContributionFactory.java:91)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.doCreate(ReflectionContributionFactory.java:60)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.create(ReflectionContributionFactory.java:42)
	at org.eclipse.fx.ui.workbench.renderers.base.BaseMenuRenderer.handleShowing(BaseMenuRenderer.java:142)
	at org.eclipse.fx.ui.workbench.renderers.base.BaseMenuRenderer.lambda$0(BaseMenuRenderer.java:80)
	at org.eclipse.fx.ui.workbench.renderers.fx.DefMenuRenderer$MenuImpl$1.handle(DefMenuRenderer.java:75)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.controls/javafx.scene.control.Menu.setShowing(Menu.java:205)
	at javafx.controls/javafx.scene.control.Menu.show(Menu.java:412)
	at javafx.controls/javafx.scene.control.skin.MenuBarSkin.showMenu(MenuBarSkin.java:471)
	at javafx.controls/javafx.scene.control.skin.MenuBarSkin.showMenu(MenuBarSkin.java:450)
	at javafx.controls/javafx.scene.control.skin.MenuBarSkin.lambda$rebuildUI$29(MenuBarSkin.java:1013)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.base/com.sun.javafx.event.EventQueue.fire(EventQueue.java:48)
	at javafx.graphics/javafx.scene.Scene$MouseHandler.handleEnterExit(Scene.java:3811)
	at javafx.graphics/javafx.scene.Scene$MouseHandler.process(Scene.java:3882)
	at javafx.graphics/javafx.scene.Scene.processMouseEvent(Scene.java:1878)
	at javafx.graphics/javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2623)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:411)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:301)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:450)
	at javafx.graphics/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:449)
	at javafx.graphics/com.sun.glass.ui.View.handleMouseEvent(View.java:557)
	at javafx.graphics/com.sun.glass.ui.View.notifyMouse(View.java:943)
	at javafx.graphics/com.sun.glass.ui.mac.MacView.notifyMouse(MacView.java:127)
Caused by: java.lang.NullPointerException: Cannot invoke "com.osgifx.console.supervisor.Supervisor.removeOSGiEventListener(com.osgifx.console.supervisor.EventListener)" because "this.supervisor" is null
	at com.osgifx.console.ui.events.handler.EventReceiveMenuContributionHandler.init(EventReceiveMenuContributionHandler.java:103)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	... 53 common frames omitted
```

--- a/com.osgifx.console.ui.events/src/main/java/com/osgifx/console/ui/events/handler/EventReceiveMenuContributionHandler.java
+++ b/com.osgifx.console.ui.events/src/main/java/com/osgifx/console/ui/events/handler/EventReceiveMenuContributionHandler.java
@@ -95,6 +95,10 @@ public final class EventReceiveMenuContributionHandler {
 
     @PostConstruct
     public void init() {
+        if (supervisor == null || supervisor.getAgent() == null) {
+            logger.atInfo().log("Agent is not connected");
+            return;
+        }
         final var currentState = Boolean.getBoolean(PROPERTY_KEY_EVENT_DISPLAY);
         if (currentState) {
             supervisor.addOSGiEventListener(eventListener);

